jsonexamples/generated/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+set(generated_files
+    utf-8.json escape.json
+    0-structurals.json 7-structurals.json 15-structurals.json 23-structurals.json
+)
+find_package(Ruby QUIET)
```

Comment:
So if ruby is missing, generated-data, will just silently be omitted, right?

What is the purpose of DEPENDS here... I think it means that CMake will know that the files in question are created by the script... but what does this dependency *do*? Does it mean that the custom target gets called automatically... or rather does it mean that targets that depend on these files will re-run?

Reply:
DEPENDS means input files: miss-templates/64.txt and miss-templates/128.txt are a series of ones and zeroes telling the generator the pattern of "branch A" vs. "branch B" blocks to generate to make branch misses most likely to happen.

OUTPUT is output files. Custom commands work just like you'd expect make targets to work: the command is re-run if the generated files are older than the output.

add_custom_target exists to let you use a *name* to trigger the generation, but it still respects depends/output.

Reply:
Ok. I understand this part:

```
  add_custom_command(
     OUTPUT ${generated_files}
     DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/miss-templates/*.txt
     COMMAND ${RUBY_EXECUTABLE} genfeaturejson.rb ${CMAKE_CURRENT_BINARY_DIR}
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   )
```

And I think I understand this part...

```
add_custom_target(generated-data DEPENDS ${generated_files})
```

But shouldn't *something* depend on generated-data?

Reply:
Right now it's just a manual target. I think benchfeatures itself should depend on generated-data, but didn't want to add the dependency because I wanted it to compile in CI. Though I bet I could make `test-data` depend on it iff RUBY is there, and make things a little nicer ...

Reply:
I added that, so at least if you do make benchfeatures when Ruby exists, it will generate the files.